siavashyj/toolbox
# feat: Create playlist downloader

# YouTube Playlist Downloader

## Description
This pull request introduces a new Python script that allows users to download information about playlists created by a YouTube account, including both private and public playlists. The script is located in the `playlist-downloader` directory of the repository.

## Features
- Authenticates with the YouTube API using OAuth 2.0
- Retrieves all playlists (both public and private) from the authenticated account
- Downloads information about all videos in each playlist
- Saves playlist information to individual text files

## Changes
- Added `playlist-downloader/main.py` as the main script
- Created `playlist-downloader/pyproject.toml` for dependency management and project configuration
- Added `playlist-downloader/README.md` with usage instructions and project information
- Included `playlist-downloader/.pre-commit-config.yaml` for code quality checks
- Included `playlist-downloader/poetry.lock` for dependency lock
- included `playlist-downloader/requirements.txt` for dependency install using pip

## How to test
1. Clone the repository
2. Navigate to the `playlist-downloader` directory: `cd playlist-downloader`
3. Install dependencies using Poetry: `poetry install`
4. Set up YouTube API credentials and save as `client_secrets.json` in the `playlist-downloader` directory
5. Run the script: `poetry run python main.py`
6. Verify that playlist information is saved to text files in the `playlist-downloader` directory

## Additional Notes
- This script requires users to set up their own YouTube API credentials
- Usage of this script should comply with YouTube's terms of service and API usage limits

## Checklist
- [x] No new warnings are generated


Please review and let me know if any changes are needed.
